### DISCUSSION TITLE
Issues with the `seq` package during installation with `use-package`

### DISCUSSION THREAD

**Discussion by nickjmeyer** [magit/magit#5276] [Support] [answered]
I'm having trouble with `seq` in `magit` with the following versions:
```
Magit 20241223.650 [>= 20241223.650], Transient 20241219.1713, Git 2.39.5, Emacs 30.0.50, gnu/linux
```

I tried starting from scratch and installing `compat` first (to avoid an issue where `static-if` was `void`).  Then installing `magit`.  Installing using `package-install` works just fine.  However, when using `use-pacakge` I run into the following error:
```
Debugger entered--Lisp error: (file-missing "Cannot open load file" "No such file or directory" $
  require(seq-25)                                                                                
  (if (version< emacs-version "25") (require 'seq-24) (require 'seq-25))                         
  eval-buffer(#<buffer  *load*-777213> nil "/home/nickjmeyer/.emacs.d/elpa/seq-2.24/seq.el" nil $
  load-with-code-conversion("/home/nickjmeyer/.emacs.d/elpa/seq-2.24/seq.el" "/home/nickjmeyer/.$
  load("/home/nickjmeyer/.emacs.d/elpa/seq-2.24/seq" nil t)                                      
  #f(compiled-function (c) #<bytecode -0x142f67cb9766a55a>)(("/home/nickjmeyer/.emacs.d/elpa/seq$
  mapc(#f(compiled-function (c) #<bytecode -0x142f67cb9766a55a>) (("/home/nickjmeyer/.emacs.d/el$
  package--reload-previously-loaded(#s(package-desc :name seq :version (2 24) :summary "Sequence$
  package-activate-1(#s(package-desc :name seq :version (2 24) :summary "Sequence manipulation f$
  package-unpack(#s(package-desc :name seq :version (2 24) :summary "Sequence manipulation funct$
  #f(compiled-function (&optional good-sigs) #<bytecode 0x1505774a6ec0369f>)((#s(epg-signature :$
  #f(compiled-function () #<bytecode -0x60b5c2a017dc62d>)()                                      
  package--with-response-buffer-1("https://elpa.gnu.org/packages/" #f(compiled-function () #<byt$
  package--check-signature("https://elpa.gnu.org/packages/" "seq-2.24.tar" "seq-2.24/\0\0\0\0\0\$
  #f(compiled-function () #<bytecode 0x1a68a968daf7ee87>)()                                      
  package--with-response-buffer-1("https://elpa.gnu.org/packages/" #f(compiled-function () #<byt$
  package-install-from-archive(#s(package-desc :name seq :version (2 24) :summary "Sequence mani$
  mapc(package-install-from-archive (#s(package-desc :name with-editor :version (20241201 1419) $
  package-download-transaction((#s(package-desc :name with-editor :version (20241201 1419) :summ$
  package-install(magit)                                                                         
  use-package-ensure-elpa(magit (t) nil)                                                         
  eval-buffer(#<buffer  *load*-762430> nil "/home/nickjmeyer/.emacs.d/setup/setup-git.el" nil t)$
  load-with-code-conversion("/home/nickjmeyer/.emacs.d/setup/setup-git.el" "/home/nickjmeyer/.em$
  require(setup-git)                                                                             
  eval-buffer(#<buffer  *load*> nil "/home/nickjmeyer/.emacs.d/init.el" nil t)  ; Reading at buf$
  load-with-code-conversion("/home/nickjmeyer/.emacs.d/init.el" "/home/nickjmeyer/.emacs.d/init.$
  load("/home/nickjmeyer/.emacs.d/init" noerror nomessage)                                       
  startup--load-user-init-file(#f(compiled-function () #<bytecode 0x1bf6823557dbb54c>) #f(compil$
  command-line()                                                                                 
  normal-top-level()                                                                             
```

Been trying to find other related issues, but not having any luck.  I'm not sure if this is is an issue with `use-package` or `magit`.  I'm guessing it might be `use-package` considering `package-install` works fine, but I figured I'd start here to see if there were any ideas first.

**Comment by tarsius** [comment]
Check if `/home/nickjmeyer/.emacs.d/elpa/seq-2.24/` contains `seq-25.el`. Also check if `/home/nickjmeyer/.emacs.d/elpa/seq-2.24/` appears in `load-path`.

> I'm guessing it might be `use-package` considering `package-install` works fine,

If I understand that correctly, once you have installed `magit` "manually", everything works fine, i.e., you can configure `magit` using `use-package`.

Yeah, to me too that sounds like `magit` is not the cause.

**Reply by nickjmeyer** [reply, marked as answer]
Here's `load-path` (or at abbreviated version).  It's in the path, though not sure why its there twice.
```
("~/.emacs.d/setup" 
...
"/home/nickjmeyer/.emacs.d/elpa/seq-2.24" "/home/nickjmeyer/.emacs.d/elpa/seq-2.24/" 
... )
```

And it does exist on disk:
```
nickjmeyer@penguin:~$ ls ./.emacs.d/elpa/seq-2.24/
seq-24.el  seq-25.el  seq-autoloads.el  seq.el  seq-pkg.el  tests
```

> If I understand that correctly, once you have installed magit "manually", everything works fine, i.e., you can configure magit using use-package.

Yes that's correct.  My current work around is to run `package-install ` when it fails and then restart emacs.  Then `use-package` will run everything as expected except the installation part.

I haven't tried installing `seq` manually.  Even though it's in my load path and exists now, I wonder if it didn't with a scratch installation which is why it failed.  Perhaps it might be a dependency issue with `use-package`.

**Comment by nickjmeyer** [comment]
It seems all I had to do was add `(use-package seq :ensure t)` before trying to install `magit`  (Not sure why I didn't think to try that before 🤦.  It's odd that `use-package` is ignoring the dependency.  Though admittedly I don't know much about package management in emacs.
